presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java

```diff
@@ -266,7 +265,13 @@ public int getPartition()
     @Override
     public TaskInfo getTaskInfo()
     {
-        return taskInfo.get();
+        return taskInfoFetcher.getTaskInfo();
```

Comment:
Right now the info is only used for stats. 
When task status transitions to a done state, we force update the taskInfo object with the done task status.